### ISSUE TITLE
Annotation GFFs with duplicate ID records

### ISSUE THREAD

**Issue by sammyjava** [legumeinfo/datastore-issues#153]
OK, I'm consolidating this issue into one since there are a lot of these GFFs with duplicate IDs. Fix 'em and check 'em off!
- [ ] Gyeongwon.gnm3.ann1
- [ ] Amiga.gnm1.ann1
- [ ] ICC4958.gnm2.ann1
- [ ] Xiabao_II.gnm1.ann1
- [ ] Zh13.gnm2.ann1
- [ ] Tifrunner.gnm1.ann1

**Comment by cann0010** [comment]
Fwiw, I am planning to start on these tomorrow (Groundhog Day; hope that doesn't augur ill)

**Comment by sammyjava** [comment]
Correct, in fact we typically have proteins and transcripts with the same ID (from different files, but mine loaders don't care about that). It's object.class + object.primaryIdentifer that must be unique (for Annotatables).

**Comment by sammyjava** [comment]
[shokin@lis ~]$ validate-annotation-collection v2/Vigna/angularis/annotations/Gyeongwon.gnm3.ann1.3Nz5/
--------------------------------------------------------------------------------
## Validating vigan collection Gyeongwon.gnm3.ann1.3Nz5
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.gene_models_main.gff3.gz
## INVALID: vigan.Gyeongwon.gnm3.ann1.3Nz5.gene_models_main.gff3.gz record ID is duplicate of one already read:
## INVALID: vigan.Gyeongwon.gnm3.SuperScaf_10	vigan.Gyeongwon.gnm3.ann1.a1	five_prime_UTR	-80053	-79912	0.0	.	ID=vigan.Gyeongwon.gnm3.ann1.Vang0010ss00060.1:five_prime_utr;Parent=vigan.Gyeongwon.gnm3.ann1.Vang0010ss00060.1
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.protein.faa.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.protein_primary.faa.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.cds.fna.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.cds_primary.fna.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.mrna.fna.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.mrna_primary.fna.gz
 x optional iprscan.gff3.gz file is not present.
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.legfed_v1_0.M65K.gfa.tsv.gz
 - vigan.Gyeongwon.gnm3.ann1.3Nz5.legfed_v1_0.M65K.pathway.tsv.gz
 x optional phytozome_10_2.HFNR.gfa.tsv.gz file is not present.

**Comment by sammyjava** [comment]
Different issue but I'll put it here since we're trying to get these annotation collections to validate.

## Validating arahy collection Tifrunner.gnm2.ann1.4K0L
 - arahy.Tifrunner.gnm2.ann1.4K0L.gene_models_main.gff3.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.protein.faa.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.protein_primary.faa.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.cds.fna.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.cds_primary.fna.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.mrna.fna.gz
 - arahy.Tifrunner.gnm2.ann1.4K0L.mrna_primary.fna.gz
 x optional iprscan.gff3.gz file is not present.
 - arahy.Tifrunner.gnm2.ann1.4K0L.legfed_v1_0.M65K.gfa.tsv.gz
## INVALID: GFA file contains gene ID arahy.Tifrunner.gnm2.ann1.B9P371 that is not present in the gene models GFF file.
 - arahy.Tifrunner.gnm2.ann1.4K0L.legfed_v1_0.M65K.pathway.tsv.gz
 x optional phytozome_10_2.HFNR.gfa.tsv.gz file is not present.

**Comment by sammyjava** [comment]
OK, that's it, seven out of nine isn't too bad.

**Comment by adf-ncgr** [comment]
OK, the issue with Gyeongwon.gnm3.ann1 should be fixed, I'd not realized on the first one that some non-CDS features were sometimes at issue as well. Will look into the remaining gfa issues, but not immediately.
